### ISSUE TITLE
How to add Progress Indicator when using YTDL inside Bash/SH

### ISSUE THREAD

**Issue by syberknight** [ytdl-org/youtube-dl#30495]
## Checklist

- [x] I'm asking a question
- [x] I've looked through the README and FAQ for similar questions
- [x] I've searched the bugtracker for similar questions including closed ones

## Question

hello,

i found some help via https://github.com/ytdl-org/youtube-dl/issues/26831 to get the kinds of downloads (custom txt, json, jpg, & mp4) that i need by using the below bash/sh script to run inside of Mac Terminal. it has been working fine (even tho it occasionally shows false errors in terminal). but over the past year it has been taking longer & longer for it to complete each time i run it (which is usually about once a month).

i read the slowdowns may be due to Youtube or ISP throttling; so i was hoping to find a way to have some sort of progress display during the process. like a progress bar or even just text percentages or something to indicate to me that the whole thing hasn't just stalled out - so i don't interrupt the process thinking there was a problem when it's just being slow.

i see that youtube-dl has an option called "--newline" which works great when putting the "youtube-dl --newline" directly into a command in Mac Terminal. but when i add it inside of this bash/sh script, it will only show all the progress at once _after_ the process is finished, instead of _during_ the download process. i've tried adding the "--newline" option after each of the "youtube-dl"'s inside the "do" section, but it just won't show during the process.

i tried other bash/sh progress methods that i could find, but i think those don't work because i don't know how to tie them into the download function (or something).

**SO, all that to say, how can i get "--newline" to work, or something similar, with this script i'm using?**

FYI: this is what i actually put into Mac Terminal...

`bash /Users/me/Sanctuary-Youtube-Backup/ytdl-dateafter.sh "https://www.youtube.com/c/SanctuaryInternationalMatrix/videos"; tput bel`

...and this is the "ytdl-dateafter.sh" content...

```
#!/bin/bash
OUTPUT_PATH="/Users/me/Sanctuary-Youtube-Backup/"
cd "$OUTPUT_PATH" || exit

FILENAME="video_ids.txt"
VIDEO_URL="$1"

# CHANGE --DATEAFTER DATE...
DATE_AFTER="20211006"

echo "$(youtube-dl --dateafter "$DATE_AFTER" --get-id "$VIDEO_URL")" > "$FILENAME"

ALL_LINES="$(cat $FILENAME)"

for VIDEO_ID in $ALL_LINES ; 
do
    LINK_STR="https://youtu.be/$VIDEO_ID"
    TITLE_TEMPLATE="%(upload_date)s %(title)s.%(ext)s"
    TITLE_TEMPLATE_NO_EXT="%(upload_date)s %(title)s"

    OUTPUT_TITLE="$(youtube-dl --dateafter "$DATE_AFTER" -o "$TITLE_TEMPLATE_NO_EXT" --get-filename "$LINK_STR")"
    sleep 2

    YOUTUBE_DL_OUTPUT="$(youtube-dl --dateafter "$DATE_AFTER" -i -o "$TITLE_TEMPLATE" -f "(bestvideo[width>=1920][ext=mp4])+bestaudio[ext=m4a]/best[ext=mp4]/best" --add-metadata --embed-thumbnail --write-thumbnail --write-info-json "$LINK_STR")"
    echo "$YOUTUBE_DL_OUTPUT"
    sleep 2

    echo "$(youtube-dl --dateafter "$DATE_AFTER" --dump-json "$LINK_STR")" | jq -r '{ Date: .upload_date | (.[0:4] + "-" + .[4:6] + "-" + .[6:8]), Title: .fulltitle, Description: .description, Tags: .tags | join(", "), Youtube: .webpage_url }' | jq -r 'to_entries | .[] | .key + ": \r" + .value + "\r\r"' > "$OUTPUT_TITLE.txt"
    sleep 2
done

rm "$FILENAME"
```

...thank you.

**Comment by kikuyan** [comment]
As to Youtube throttling, you can find several issue reports, and there's a PR #30184.
Or simply try yt-dlp.

**Comment by syberknight** [comment]
i guess i'm not getting notified when someone replies (another issue to troubleshoot:-(

so, i'm really sorry but i just don't understand what to do. i read the stackexchange page you gave a link for but it's just way over my head.

would REALLY appreciate you or someone really spelling it out for me on what to do & how to do it.

someone else created the above script for me, but i can't reach them for help on it or i would.

**Comment by syberknight** [comment]
ack! it looks like this issue has been closed :-(

please reopen it.
